### PR TITLE
Fixing bug that started playback at 2nd segment

### PR DESCRIPTION
**Pull request by bc-bbay**
Bugfix and unit test

### CODE REVIEW

**Comment by bclwhitaker** [comment]
:+1:

File: test/videojs-hls_test.js [hunk 1]
```diff
@@ -1724,6 +1724,41 @@ test('calling play() at the end of a video resets the media index', function() {
...
+  var oldMedia, newMedia, compareBuffer;
+  player.src({
+    src: 'https://example.com/encrypted-media.m3u8',
+    type: 'application/vnd.apple.mpegurl'
+  });
+  openMediaSource(player);
+
+  oldMedia = player.hls.playlists.media;
+  newMedia = {segments: [{
+    key: {
```
Comment (dmlap) [inline test/videojs-hls_test.js]: Are the key properties important to this test?

File: test/videojs-hls_test.js [hunk 2]
```diff
@@ -1724,6 +1724,41 @@ test('calling play() at the end of a video resets the media index', function() {
...
+  player.hls.playlists.media = function() {
+    return newMedia;
+  };
+
+  player.hls.sourceBuffer = undefined;
+  compareBuffer = [{mediaIndex: 0, playlist: newMedia, offset: 0, bytes: [0,0,0]}];
+  player.hls.segmentBuffer_ = [{mediaIndex: 0, playlist: newMedia, offset: 0, bytes: [0,0,0]}];
+
+  player.hls.drainBuffer();
+  deepEqual(player.hls.segmentBuffer_,compareBuffer,'playlist remains unchanged');
```
Comment (dmlap) [inline test/videojs-hls_test.js]: Please add whitespace between arguments to functions.

File: test/videojs-hls_test.js [hunk 3]
```diff
@@ -1724,6 +1724,41 @@ test('calling play() at the end of a video resets the media index', function() {
...
+    },
+    uri: 'http://media.example.com/fileSequence53-B.ts'
+  }]};
+  player.hls.playlists.media = function() {
+    return newMedia;
+  };
+
+  player.hls.sourceBuffer = undefined;
+  compareBuffer = [{mediaIndex: 0, playlist: newMedia, offset: 0, bytes: [0,0,0]}];
+  player.hls.segmentBuffer_ = [{mediaIndex: 0, playlist: newMedia, offset: 0, bytes: [0,0,0]}];
```
Comment (dmlap) [inline test/videojs-hls_test.js]: If you could add a comment explaining why this comparison proves that drainBuffer() isn't dropping segments, I think that would make this test easier to understand.

**Comment by bc-bbay** [comment]
Making the segment a key makes the bug-infected code throw an error sooner. It gets to the segmentBuffer.shift() without having to override other function calls that happen in drainBuffer() and are irrelevant to the test, and I think it's the lesser of two evils.

File: test/videojs-hls_test.js (hunk 1 above)
Comment (bc-bbay) [inline test/videojs-hls_test.js]: Making the segment a key makes the bug-infected code throw an error sooner. It gets to the segmentBuffer.shift() without having to override other function calls that happen in drainBuffer() and are irrelevant to the test, and I think it's the lesser of two evils.

File: test/videojs-hls_test.js (hunk 2 above)
Comment (bc-bbay) [inline test/videojs-hls_test.js]: Will do.

File: test/videojs-hls_test.js (hunk 3 above)
Comment (bc-bbay) [inline test/videojs-hls_test.js]: Will do.

**Pull request merged**